torch/distributed/_tensor/device_mesh.py

```diff
@@ -107,6 +107,9 @@ def __init__(
         _init_process_groups: bool = True,
     ) -> None:
         self.device_type = device_type
+        self._device_handle = (
```

Comment:
I don't quite like the `device_handle` thing, more like a hack to me specialized to cuda. if you are inventing a new backend that conforms with cuda, shouldn't it also have a identical call to `torch.cuda.set_device`, or maybe we can use `CUDA_VISIBLE_DEVICES` if it's hard to hijack into `torch.cuda`

Reply:
> if there's additional gaps to make your case work in this PR?

We hope that the distributed nature of PyTorch can support third-party devices and third-party backends, not just third-party backends.

In fact, our device and cuda have the same semantics, so it will also be consistent with the cuda interface, which is a natural approach. For example, xpu uses `torch._register_device_module("xpu", current_module)` to register its extension to `torch.xpu`, and then we can use `torch.xpu.device_count()` and `torch.xpu.get_rng_state()` realize the same function as in cuda. Our implementation is similar to xpu, so for this device using `torch.custom_device`, we want use `_device_handle` instead of directly using `torch.cuda` to support cuda-like devices which like in FSDP.

Reply:
I see, this make sense then, could you rebase the PR and fix the merge conflict?

Reply:
I rebase the PR. In addition, the situation that `init_process_group` does not initialize custom_backend by default can be discussed and resolved in the next PR.